src/NuGet.Core/NuGet.DependencyResolver.Core/GraphModel/GraphOperations.cs

```diff
@@ -31,11 +31,38 @@ public static AnalyzeResult<RemoteResolveResult> Analyze(this GraphNode<RemoteRe
             root.TryResolveConflicts(result.VersionConflicts);
 
             // Remove all downgrades that didn't result in selecting the node we actually downgraded to
-            result.Downgrades.RemoveAll(d => d.DowngradedTo.Disposition != Disposition.Accepted);
+            result.Downgrades.RemoveAll(d => !IsRelevantDowngrade(d));
 
             return result;
         }
 
+        /// <summary>
+        /// A downgrade is relevant if the node itself was `Accepted`.
+        /// A node that itself wasn't `Accepted`, or has a parent that wasn't accepted is not relevant.
+        /// </summary>
+        /// <param name="d">Downgrade result to analyze</param>
+        /// <returns>Whether the downgrade is relevant.</returns>
+        private static bool IsRelevantDowngrade(DowngradeResult<RemoteResolveResult> d)
+        {
+            return d.DowngradedTo.Disposition == Disposition.Accepted && AreAllParentsAccepted(d);
+
+            static bool AreAllParentsAccepted(DowngradeResult<RemoteResolveResult> d)
+            {
+                var resultToCheck = d.DowngradedFrom.OuterNode;
```

Comment:
Is this outer node which caused downgrade only one? Currently here only find one then bubble up to top searching. In below case is it bubble up in A or B path? Or it check both of them?

    A 1.0.0 -> F 1.0.0-> C 1.0.0 -> D 1.1.0
    B 1.0.0 -> C 1.0.0 -> D 1.1.0
    E 1.0.0 -> C 1.1.0 -> D 1.0.0
    D 1.0.0

Reply:
Use explicit type.

Reply:
@erdembayar 
In that scenario, there'll be 2 downgrade checks, because there are 2 different paths to D 1.1.0.